pkg/controller/daemon/util/daemonset_util.go

```diff
@@ -133,3 +136,48 @@ func SplitByAvailablePods(minReadySeconds int32, pods []*v1.Pod) ([]*v1.Pod, []*
 	}
 	return availablePods, unavailablePods
 }
+
+// AppendDaemonSetPodNodeAffinity adds a NodeAffinity term with the given "nodeName" to the "affinity" terms.
+func AppendDaemonSetPodNodeAffinity(affinity *v1.Affinity, nodename string) *v1.Affinity {
```

Comment:
Where's the caller of this function? Prefer adding a new function with its caller. 

Should we check if the given `affinity` already has a hostname set?